src/query/storages/fuse-meta/src/meta/common.rs

```diff
@@ -37,6 +39,53 @@ pub struct ColumnStatistics {
     #[serde(alias = "unset_bits")]
     pub null_count: u64,
     pub in_memory_size: u64,
+    pub hll: Option<HyperLogLog<DataValue>>,
+}
+
+impl ColumnStatistics {
+    pub fn new(min: DataValue, max: DataValue, null_count: u64, in_memory_size: u64) -> Self {
+        ColumnStatistics {
+            min,
+            max,
+            null_count,
+            in_memory_size,
+            hll: Some(HyperLogLog::new(0.00408)),
+        }
+    }
+
+    pub fn new_empty() -> Self {
+        ColumnStatistics {
+            min: DataValue::Null,
+            max: DataValue::Null,
+            null_count: 0,
+            in_memory_size: 0,
+            hll: Some(HyperLogLog::new(0.00408)),
+        }
+    }
+
+    pub fn add(&mut self, value: &DataValue) {
+        self.hll.as_mut().unwrap().push(value);
+    }
+
+    pub fn hll(&self) -> &HyperLogLog<DataValue> {
+        self.hll.as_ref().unwrap()
+    }
+
+    pub fn merge(&mut self, other: &ColumnStatistics) {
```

Comment:
not sure if `merge` should be a method of ColumnStatistics:
-  this method merges hll only
-  `ColumnStatistics` is more likely to be a value object

Reply:
rename `merge` to `merge_number_of_distinct_values` to be more clearly.